sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerReadIT.java

```diff
@@ -61,6 +62,12 @@
 
   /** Pipeline options for this test. */
   public interface SpannerTestPipelineOptions extends TestPipelineOptions {
+    @Description("Project that hosts Spanner instance")
+    @Nullable
+    String getInstanceProjectId();
```

Comment:
I am very slightly concerned, because pipeline options are a global namespace and this option does not have "spanner" in the name.

Reply:
Seems like this is only for integration tests ?

Reply:
Yes. Do pipeline options defined in ITs pollute the global namespace, @kennknowles ? 
If so, there are other options in this and other tests that this concern applies to.

Reply:
As long as the PipelineOption interface isn't being registered via a ServiceLoader automatically, a call to PipelineOptionsFactory.resetCache() after each test in a `@TearDown` method will remove the pollution of the namespace. If it is being registered via serviceloader then people have to be careful around name choices/classpaths.

Reply:
We call https://github.com/apache/beam/blob/ef67378fec2afadbe4a5303e61ac0427ced1faa0/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerReadIT.java#L98 
But I think, we don't have a registrar for these options, such as, say: https://github.com/apache/beam/blob/ef67378fec2afadbe4a5303e61ac0427ced1faa0/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpPipelineOptionsRegistrar.java#L26